gramps/gui/views/treemodels/citationtreemodel.py

```diff
@@ -211,17 +212,23 @@ def add_row2(self, handle, data):
         # add the citation as a child of the source. Otherwise we add the source
         # first (because citations don't have any meaning without the associated
         # source)
-        if self._get_node(data[5]):
+        if self._get_node(data["source_handle"]):
```

Comment:
Pull "source_handle" out as a constant and use here and line 222 below

Reply:
See note.